### PR TITLE
Final Thee Worth App Project

### PR DESCRIPTION
**Pull request by hassanShakur**
The following are the significant changes that occurred in this and the other 2 branches:

- I picked the [Financial modeling prep API](https://financialmodelingprep.com/developer/docs/) for stock data.
- Made the website design based on an Original design idea by [Nelson Sakwa on Behance](https://www.behance.net/sakwadesignstudio).

- On the `homepage`:
    - When the page loads, the web app shows the stock companies that could be filtered out by name through the search bar.
    - Displayed some numerics along with the companies.
    - The application navigates to the details page when the user clicks (or taps) on a company.

- On the `Details page`:
    - the web app retrieves data from the API to show detailed stock information of the company.
    - When the user clicks on the `Back` button (<), the user navigates to the home page.

- I also added some tests for the redux store and components.

### CODE REVIEW

File: src/App.css [hunk 1]
```diff
@@ -8,12 +8,12 @@
   margin: 0;
```
Comment (Alfredbis29) [inline src/App.css:8]: - [ ] I just noticed that you've done a great job so far on this project, and I commend your efforts :clap:. However, I'd like to address some feedback regarding the design aspect of the application. 
- [ ] While the functionalities are well-implemented, I noticed that the design does not entirely align with the specified guidelines provided by Microverse Team. In particular, I recommend focusing on refining the typography, layout, and color scheme to better match the [guidelines Design](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)) . By doing so, we can create a more cohesive and user-friendly experience. Your dedication to improvement is admirable, and I'm confident that with some adjustments, the final result will be outstanding. Keep up the excellent work! If you need any support or guidance, feel free to reach out. :wink:  

- [ ] Creating another video presentation with the updated design will provide valuable practice and help ensure a polished user experience. Your dedication to improvement is admirable, and I'm confident the final outcome will be exceptional. kindly fix this :point_down: 
Screenshot from 2023-08-02 14-37-25

Choose the design among these two :point_down:  this how it should look like :point_down: 

https://github.com/microhass/thee-worth/assets/88894525/86eea6c8-0275-43cd-9ec1-5ae2a8630d53

**Review by Alfredbis29 — CHANGES_REQUESTED** [review]
#  STATUS: REQUIRED CHANGES 
 ## Hi @hassanShakur 
 Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

  
 ## Highlights 
:heavy_check_mark:  Your Work looks Professional 
:heavy_check_mark:  Nice work so far :+1: 
 
 ## Required Changes 

- [ ]  Check Under the review

 ## Optional suggestions 

- [ ] N/A
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._
 
 Cheers and Happy coding! :clap: :clap: :clap:
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @Alfredbis29 in your question so I can receive the notification.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/companies/index.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+  if (queryResults?.length === 0) { return <div className="loading">No results found!</div>; }
+
+  return (
+    <main>
+      {queryResults.map((company, i) => (
+        <Company
+          key={company.symbol.concat(i)}
+          company={company}
+        />
```
Comment (thecodechaser) [inline src/components/companies/index.js:29]: - [ ] As requested by the previous reviewer to follow the design guidelines, Please fix the below design issues.
  - Please make sure to display two items(companies) in a row. Currently, It's displaying only one item.
  - Please make sure to display a small button on each item as on the given design, That will indicate users going to the details page.

### Your design

image

### Given design

image

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @hassanShakur👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## To Highlight 🎉
- Highly organized code. ✔️ 
- Nice UI design. ✔️
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- No linter errors. :heavy_check_mark:
- PR has a good title and summary. ✔️



 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by joel-kalema — APPROVED** [review]
Hi @hassanShakur 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# Congratulations! 🎉 


## Highlights

The project is running successfully 👍 
PR is well descriptive 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@joel-kalema** in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**